### ISSUE TITLE
C# extension or C# Dev Kit extension?

### ISSUE THREAD

**Issue by Demkeys** [dotnet/vscode-csharp#6054]
Long story short, yesterday intellisense stopped working for me. I already had C# extension installed. And for whatever reason suddenly the Restart Omnisharp option was gone too. Then I noticed the C# Dev Kit extension in the extensions list and decided to install it. That fixed the intellisense issue. 
Considering this, I would like to try and understand what changed that caused intellisense to stop working. 
But also, is C# Dev Kit extension the preferred way to go now, since it also installs C# ext and Intellicode for C# Dev Kit ext (and considering that intellisense only started working after that)?

**Comment by Kidsunbo** [comment]
Same question, and now the problem is that C# extension doesn't work like before.

**Comment by DerGary** [comment]
This is interesting for me too, the main reason not to switch to C# Dev Kit is the Licencing:
> C# Dev Kit builds on the same foundations as Visual Studio for some of its functionality, it uses the same license model as Visual Studio. This means it's free for individuals, as well as academia and open-source development, the same terms that apply to Visual Studio Community. **For organizations, the C# Dev Kit is included with Visual Studio Professional and Enterprise subscriptions, as well as GitHub Codespaces**. 

So it is not free of charge for organizations to use the extension although the C# Extension previously was for free and contained all the necessary features. Now its unusable anymore without the C# Dev Kit which costs money. I don't like this development

**Comment by arkalyanms** [comment]
The C# extension should still be functional without the C# Dev Kit and Intellicode addons. 

If something is broken in C#, it is a likely a bug and not by design. Can we please get the repro and the C# logs so we can fix it asap?

**Comment by Demkeys** [comment]
The best way I could think of for creating a repro was to uninstall all the related extensions so I uninstalled C#, C# Dev Kit, Intellicode for C# Dev Kit and Unity extensions, then reinstalled C# extension only.
Here's the repro: 
* Make sure you only have C# extension (not C# Dev Kit or Intellicode for C# Dev Kit).
* Create new C# Console project using ``` dotnet new console ``` .
* Open ```Program.cs``` and type in some code. See if Intellisense works.
* Additionally, if it does happen to work with a new project, close VSCode and open it again and see if it works after that.
In my case it didn't work at all, upon project creation or even after restarting VSCode.

Here is some info about the tools:
* C# extension: v2.0.328
* .NET SDK: 7.0.306

Here are the Extension logs:
```
2023-08-07 20:50:27.153 [info] Extension host with pid 19628 started
2023-08-07 20:50:27.154 [info] Skipping acquiring lock for c:\Users\Demkeys\AppData\Roaming\Code\User\workspaceStorage\af198d7ebed9aad432609990f980b58c.
2023-08-07 20:50:27.298 [info] ExtensionService#_doActivateExtension ms-dotnettools.vscode-dotnet-runtime, startup: false, activationEvent: 'onLanguage:csharp', root cause: ms-dotnettools.csharp
2023-08-07 20:50:27.396 [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onLanguage'
2023-08-07 20:50:27.486 [info] ExtensionService#_doActivateExtension ms-dotnettools.csharp, startup: false, activationEvent: 'onLanguage:csharp'
2023-08-07 20:50:28.647 [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.git
2023-08-07 20:50:28.658 [info] ExtensionService#_doActivateExtension vscode.github-authentication, startup: false, activationEvent: 'onAuthenticationRequest:github'
2023-08-07 20:50:28.979 [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
2023-08-07 20:50:29.078 [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
2023-08-07 20:50:29.326 [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication, startup: false, activationEvent: 'onAuthenticationRequest:microsoft'
2023-08-07 20:50:29.576 [info] Eager extensions activated
2023-08-07 20:50:29.952 [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: false, activationEvent: 'onStartupFinished'
2023-08-07 20:50:29.961 [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
2023-08-07 20:50:29.998 [info] ExtensionService#_doActivateExtension ms-vscode-remote.remote-wsl-recommender, startup: false, activationEvent: 'onStartupFinished'

```

**Comment by JustinGrote** [comment]
Probably dupe of: https://github.com/dotnet/vscode-csharp/issues/6024, see the current workaround there for what is clearly an issue with the 2.0 release.

**Comment by dibarbet** [comment]
> Create new C# Console project using dotnet new console .
Open Program.cs and type in some code. See if Intellisense works.
Additionally, if it does happen to work with a new project, close VSCode and open it again and see if it works after that.
In my case it didn't work at all, upon project creation or even after restarting VSCode.

Thanks for the steps - this issue is at least known, currently the 2.x version requires a solution file until https://github.com/dotnet/vscode-csharp/issues/5722 is fixed.

**Comment by Demkeys** [comment]
I just tested this out, and you're right. I created a new console project while having only C# extension installed, no sln file created. Then when I install C# Dev Kit and restart the language server, a sln file gets created. 
If I want this to work using only C# extension I have to create the sln file manually and add the current project to it.
```
dotnet new sln
dotnet new console
dotnet sln add [ProjectName].csproj --in-root 
```
After that I restart the language server and intellisense works again.
That being said, while this issue is being worked on, I think my options are:
* Enable ```dotnet.server.useOmnisharp``` flag and restart VSCode. This way I can still use C# ext v2.0.
* Revert back to the pre-2.0 C# extension version.
* Use the method where I manually create the sln file and add the project. I've never had to do this before, so I don't know if there could be any potential drawbacks of doing it manually.

Thanks for the info. Closing this issue now.
